styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/FlyteRunnerHandler.java

```diff
@@ -84,4 +124,43 @@ public void safeTransitionInto(final RunState state, final EventRouter eventRout
         // do nothing
     }
   }
+
+  /**
+   * Map Styx id to Flyte exec names.
+   *
+   * Flyte exec names should be 20 characters long and it have conform to DNS-1123 subdomain,
+   * however for some reason it must start with a lowercase alphabetic character.
+   */
+  static class StyxIdToFlyteExecNameMapper implements Function<String, String> {
```

Comment:
i dont think the static class/function adds much here. I believe it would be easier to navigate as a separate class instead.